docs/gc_overview.md

```diff
@@ -23,81 +23,183 @@
+
+### GC *classunloading* operation
+
+Not parallel, single threaded
+
+In a verbose GC log, this operation is shown by the `<gc-op type="classunload">` XML element. For more information, see [Verbose GC logs](vgclog.md).
+
+
+### GC *compact* operation
+
+Compaction of the heap is an expensive operation because when objects are moved to defragment the heap, the references to each object change. Therefore, *compact* operations do not occur by default but only when the following triggers occur:
+
+- The `-Xcompactgc` option is specified on the command line.
+- After sweeping the heap, there is not enough free space available to satisfy an allocation request.
+- A `System.gc()` is requested and the last allocation failure that triggered a global GC cycle did not compact or `-Xcompactexplicitgc` is specified.
+- At least half the previously available memory has been consumed by TLH allocations (ensuring an accurate sample) and the average TLH size falls to less than 1024 bytes.
+- The largest object that the `gencon` GC policy failed to move to the *tenure* area in the most recent *scavenge* operation is bigger than the largest free slot in the *tenure* area.
+- The heap is fully expanded and less than 4% of the *tenure* area is free.
+- Less than 128 KB of the heap is free.
+
+The following two options can be used to control compaction:
+
+- [`-Xcompactgc`](xcompactgc.md) forces compaction of the heap.
+- [`-Xnocompactgc`](xnocompactgc.md) avoids compaction of the heap as a result of all the triggers shown in the preceding list. However a compaction can still occur in rare circumstances.
+
+In a verbose GC log, this operation is shown by the `<gc-op type="compact">` XML element. For more information, see [Verbose GC logs](vgclog.md).
+
+## Weak reference processing
+
+Weak reference processing includes *soft* references, *weak* references, and *phantom* references. These references are created by the user for specific use cases and allow
+some level of interaction with the garbage collector. For example, a *soft* reference to an object allows that object to persist in memory for a longer period of time before being
+cleared. For example, a software cache. The garbage collector handles the reference types in the order shown and with the behavior detailed in the following table:
+
+
+| Reference type | Class                            | Garbage collector behavior |
+|----------------|----------------------------------|----------------------------|
+| *soft*         | `java.lang.ref.SoftReference`    | A *soft* reference is cleared only when its *referent* has not been marked for a number of GC cycles or if space on the heap is likely to cause an out of memory error. Use the [-Xsoftrefthreshold](xsoftrefthreshold.md) option to control the collection of *soft* reference objects. |
+| *weak*         | `java.lang.ref.WeakReference`    | A *weak* reference is cleared as soon as the object it references is removed. |
```

Comment:
Try to align the terminology....

In Soft you use 'its referent' and weak you use 'the object it references' - indeed same thing, but maybe not obvious to every one.
In Soft you say 'has not been marked' and in Weak you say 'removed' The former is more correct for both. We don't really remove it, although we'll likely recycle its space by the end of this cycle/ at the moment of reference processing, we just know that it has not been marked as live (or more specifically not copy-forwarded or not scavenged for evacuating style of GCs).

In s

Note also that weak refs behave almost like Soft with threshold 0.